### PR TITLE
Allow python-dtrace to work on macOS

### PR DESCRIPTION
**Pull request by arichardson**
Buffered libdtrace output on macOS appears to be broken until at least 10.15:

The -B flag is disabled in dtrace.c (https://opensource.apple.com/source/dtrace/dtrace-338.40.5/cmd/dtrace/dtrace.c.auto.html)
and compiling the dtrace command line tool with that flag enabled results in the same crash
that I see with python-dtrace:
```
* thread 1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x7fb100000000)
  * frame 0: 0x00007fff6f83de52 libsystem_platform.dylib`_platform_strlen + 18
    frame 1: 0x00007fff6f6d9891 libsystem_c.dylib`__vfprintf + 5379
    frame 2: 0x00007fff6f6ffad3 libsystem_c.dylib`__v2printf + 475
    frame 3: 0x00007fff6f6e5ee7 libsystem_c.dylib`_vsnprintf + 417
    frame 4: 0x00007fff6f6e5f90 libsystem_c.dylib`vsnprintf + 68
    frame 5: 0x00007fff6d3650e3 libdtrace.dylib`dt_printf + 524
    frame 6: 0x00007fff6d33da7f libdtrace.dylib`dt_consume_cpu + 2536
    frame 7: 0x00007fff6d33c9e0 libdtrace.dylib`dtrace_consume + 1090
    frame 8: 0x00007fff6d366534 libdtrace.dylib`dtrace_work + 116
    frame 9: 0x000000010d6be9ff dtrace2`main(argc=4, argv=0x00007ffee2547250) at dtrace2.c:1834:17
    frame 10: 0x00007fff6f647cc9 libdyld.dylib`start + 1
```

Work around this by passing a FILE * obtained from open_memstream
and reading it after each call to dtrace_work().
This should generally result in the same output (the hello world test now
passes on macOS), but there could be some corner cases that are different.

### CODE REVIEW

**Comment by arichardson** [comment]
Note: the buffered output callback seems to be broken since at least macOS 1.6.4 (https://stackoverflow.com/questions/3792486/libdtrace-buffered-output)

File: dtrace_ctypes/consumer.py [hunk 1]
```diff
@@ -284,7 +310,6 @@ def run(self, script, runtime=1):
             raise Exception('Failed to walk the aggregate: ',
                             get_error_msg(self.handle))
 
-
```
Comment (arichardson) [inline dtrace_ctypes/consumer.py:287]: Accidental change, will force push to remove.

**Comment by tmetsch** [comment]
Did a quick test on Solaris, and this currently breaks a clean build there. Need to investigate a bit more, might be tomorrow, though :-)

**Comment by arichardson** [comment]
Sorry about that. The latest version works on macOS+FreeBSD for me.

**Comment by tmetsch** [comment]
Ok a clean build seems to work, will merge this in a bit. Do you think we are good to make a new release?

**Comment by arichardson** [comment]
> Ok a clean build seems to work, will merge this in a bit. Do you think we are good to make a new release?

I found two more small bugs while testing the Cython consumer. Releasing once this PR is merged and those two bugs are fixed sounds good to be.

**Comment by tmetsch** [comment]
> 
> 
> > Ok a clean build seems to work, will merge this in a bit. Do you think we are good to make a new release?
> 
> I found two more small bugs while testing the Cython consumer. Releasing once this PR is merged and those two bugs are fixed sounds good to be.

Ok, sounds like a plan - thanks for all your effort!

**Comment by arichardson** [comment]
> > > Ok a clean build seems to work, will merge this in a bit. Do you think we are good to make a new release?
> > 
> > 
> > I found two more small bugs while testing the Cython consumer. Releasing once this PR is merged and those two bugs are fixed sounds good to be.
> 
> Ok, sounds like a plan - thanks for all your effort!

No problem, thanks for reviewing and testing so quickly! I've now published all my local changes as pull requests.
